v3/lock.go

```diff
@@ -28,7 +28,7 @@ import (
 type Lock struct {
 	semaphore sync.Mutex
 
-	client       *Client
+	client       *internalClient
```

Comment:
Isn't the sort key an attribute (even if indirect) of the Lock? In other words, would it be necessary splitting `Lock` into two types?

Reply:
The way I was going about implementing it was that `internalClient` has a `sortKeyName` field, which for normal `Client`s is empty (but for `ClientWithSortKey` is not). This way, methods on both clients (eg `AcquireLockWithContext`) will simply route to an internal function `acquireLockWithContext`, with `Client`'s passing an empty string for the `sortKey`. And thus, the only distinction between the two types is simply the interface they provide to the user.

Reply:
Got it